QualityControl/lib/repositories/LayoutRepository.js

```diff
@@ -24,16 +24,31 @@ export class LayoutRepository extends BaseRepository {
    * @param {number} [options.owner_id] - Filter layouts by owner ID
    * @param {string} [options.name] - Filter layouts by exact name match
    * @param {Array<string>} [options.fields] - Array of field names to include in each returned layout object
+   * @param {string} [options.filter] - Filter layouts by containing filter.objectPath, case insensitive
    * @returns {Array<object>} Array of layout objects matching the filters, containing only the specified fields
    * @throws {TypeError} If fields parameter is provided but is not an array
    * @throws {Error} If any specified field does not exist in the layout objects
    */
-  listLayouts({ name, owner_id, fields = [] } = {}) {
+  listLayouts({ name, owner_id, fields = [], filter } = {}) {
     const { layouts } = this._jsonFileService.data;
+    const filterPresent = filter !== undefined;
 
-    const layoutFilter = (layout) =>
-      (owner_id === undefined || layout.owner_id === owner_id) &&
-      (name === undefined || layout.name === name);
+    // Better to lowercase once rather than at every single object_path comparison.
+    if (filterPresent && filter.objectPath !== undefined) {
```

Comment:
I would recommend you make use of the optional chaining in this instance for improved readability. Something along the lines:
```js
if (filter?.objectPath !== undefined) {
  // ...
}
```
Same for the lines below